### PR TITLE
Support Reviewdog Diagnostic Format

### PR DESCRIPTION
**Pull request by haya14busa**
It makes the diagnostic reports richer and addes link to a rule URL for
example with -github-pr-review reporter.

Reviewdog Diagnostic Format: https://github.com/reviewdog/reviewdog/tree/master/proto/rdf

### CODE REVIEW

File: to-rdjson.jq [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Convert hadlint JSON output to Reviewdog Diagnostic Format (rdjson)
```
Comment (mgrachev) [inline to-rdjson.jq:1]:
```suggestion
# Convert hadolint JSON output to Reviewdog Diagnostic Format (rdjson)
```

File: to-rdjson.jq [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    url: "https://github.com/hadolint/hadolint"
+  },
+  diagnostics: . | map({
+    message: .message,
+    code: {
+      value: .code,
+      url: (if .code | startswith("DL") then
+              "https://github.com/hadolint/hadolint/wiki/\(.code)"
+            elif .code | startswith("SC") then
+              "https://github.com/koalaman/shellcheck/wiki/\(.code)"
```
Comment (mgrachev) [inline to-rdjson.jq:15]: I think it came from another linter 🙂

**Review by mgrachev — COMMENTED** [review]
Thank you! 🔥

File: to-rdjson.jq (hunk 1 above)
Comment (haya14busa) [reply to-rdjson.jq]: Oops, thank you for the catch!

File: to-rdjson.jq (hunk 2 above)
Comment (haya14busa) [reply to-rdjson.jq:15]: Yes... and it's expected. See https://github.com/hadolint/hadolint

Comment (mgrachev) [reply to-rdjson.jq:15]: I got it. Thank you.

**Review by mgrachev — APPROVED** [review]
👍

**Pull request merged**